samcli/commands/_utils/options.py

```diff
@@ -94,6 +94,22 @@ def guided_deploy_stack_name(ctx, param, provided_value):
     return provided_value if provided_value else DEFAULT_STACK_NAME
 
 
+def image_repositories_callback(ctx, param, provided_value):
+    """
+    Create an dictionary of function logical ids to ECR URIs.
+    :param ctx: Click Context
+    :param param: Param name
+    :param provided_value: Value provided by Click, after being processed by ImageRepositoriesType.
+    :return: dictionary of function logic ids to ECR URIs.
+    """
+
+    image_repositories = {}
+    for value in provided_value:
+        image_repositories.update(value)
+
+    return image_repositories if image_repositories else None
```

Comment:
Why do we returning `None` instead of empty dict? Asking because this might cause errors down the execution where we call operations on the dict.

Reply:
I had this because of previous validation issues when trying to do validation in the respective callbacks, we should just be able to return dict, now.